share/sysbuild/images/bootloader/Kconfig

```diff
@@ -147,7 +147,7 @@ config BOOT_SIGNATURE_KEY_FILE
 
 config BOOT_ENCRYPTION
 	bool "Encrypted image support"
-	depends on !BOOT_SIGNATURE_TYPE_NONE
+	depends on !BOOT_SIGNATURE_TYPE_NONE && !MCUBOOT_MODE_DIRECT_XIP && !MCUBOOT_MODE_DIRECT_XIP_WITH_REVERT && !MCUBOOT_MODE_FIRMWARE_UPDATER
```

Comment:
perhaps this is a good time to introduce a helper symbol, for example `SUPPORT_BOOT_ENCRYPTION`.
`SUPPORT_BOOT_ENCRYPTION` can then either default based on above conditions, or be selected by the appropriate MCUboot settings, like signature type under certain conditions.

Reply:
Updated